src/containers/BubbleEditor/BubbleEditor.js

```diff
@@ -41,13 +44,95 @@ class BubbleEditor extends React.Component {
         _timePoints.add(bubble[RANGE.END_TIME]);
         return _timePoints;
       }, new Set())
+    ).sort((p1, p2) => p1 - p2);
+
+  clearTextSelection = () => {
+    if (window.getSelection) {
+      if (window.getSelection().empty) {
+        // Chrome
+        window.getSelection().empty();
+      } else if (window.getSelection().removeAllRanges) {
+        // Firefox
+        window.getSelection().removeAllRanges();
+      }
+    } else if (document.selection) {
+      // IE?
+      document.selection.empty();
+    }
+  };
+
+  dragStart = ev => {
+    if (
+      ev.target === ev.target.parentNode.firstChild ||
+      ev.target === ev.target.parentNode.lastChild
+    ) {
+      return;
+    }
+    const selectedPoint = Array.prototype.indexOf.call(
+      ev.target.parentNode.childNodes,
+      ev.target
     );
 
+    document.body.addEventListener('mousemove', this.dragMove);
+    document.body.addEventListener('mouseup', this.dragEnd);
```

Comment:
Is there an alternative to attaching to the body? We could grab the ref off of the div in this component, but that would limit the dragging to the component. Hmm. Tricky, this is breaking the component encapsulation a little.